spec/requests/api/v1/chats/show_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'rails_helper'
+
+RSpec.describe 'GET /api/v1/chats/:id', type: :request do
+  subject { get api_v1_chat_url(param), headers: headers, as: :json }
+
+  let!(:user) { create(:user) }
+  let!(:chat) { create(:chat, user_a: user) }
+
+  context 'with logged-in user' do
+    let(:headers) { auth_headers }
+
+    context 'when the request is valid and the chat has messages' do
+      let(:param) { chat.id }
+
+      context 'when the chat has messages' do
+        let!(:messages) { create_list(:message, 3, chat: chat) }
+
+        it 'returns the chat and the messages as a json' do
+          subject
+          chat_messages = messages.map do |message|
+            {
+              user_id: message.user.id,
+              text: message.text
```

Comment:
You could also add the `message.created_at` attribute, so that the user could see the datetime of each message.